### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

File: lib/vsc/utils/nagios.py [hunk 1]
```diff
@@ -262,7 +262,9 @@ def cache(self, nagios_exit, nagios_message):
         try:
             p = pwd.getpwnam(self.nagios_username)
             os.chmod(self.filename, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IWGRP)
-            os.chown(self.filename, p.pw_uid, p.pw_gid)
+            # only change owner/group when run as root
+            if os.geteuid() == 0:
```
Comment (stdweird) [inline lib/vsc/utils/nagios.py]: and also warn/error if you are trying to do this as non-root

**Comment by stdweird** [comment]
@wpoely86 version bump

File: lib/vsc/utils/timestamp_pid_lockfile.py [hunk 2]
```diff
@@ -93,12 +94,12 @@ def acquire(self):
...
-                        raise LockFailed
-                else:
-                    self.logger.error('Unable to obtain lock on %s' % (self.path))
-                    raise LockFailed
+                        doraise = False
+                    except:
+                        pass
+            if doraise:
+                self.logger.error('Unable to obtain lock on %s: %s' % (self.path, err))
+                raise LockFailed
```
Comment (JensTimmerman) [inline lib/vsc/utils/timestamp_pid_lockfile.py]: you actually replaced 6 lines of code with 7 new ones and made the flow less clear

Comment (stdweird) [inline lib/vsc/utils/timestamp_pid_lockfile.py]: as opposed to waht? the code now does: always raise unless some condition

Comment (JensTimmerman) [inline lib/vsc/utils/timestamp_pid_lockfile.py]: as apposed to how it was before?

Comment (stdweird) [inline lib/vsc/utils/timestamp_pid_lockfile.py]: the flow before was so obvious that it ignored a whole range of exceptions?

File: lib/vsc/utils/timestamp_pid_lockfile.py [hunk 3]
```diff
@@ -84,6 +84,7 @@ def acquire(self):
             _write_pid_timestamp_file(self.path)
             self.logger.info('Obtained lock on timestamped pid lockfile %s' % (self.path))
         except OSError, err:
```
Comment (JensTimmerman) [inline lib/vsc/utils/timestamp_pid_lockfile.py]: use `except OSError as err`

File: lib/vsc/utils/timestamp_pid_lockfile.py (hunk 2 above)
Comment (JensTimmerman) [inline lib/vsc/utils/timestamp_pid_lockfile.py]: so this changes the behaviour? in that case, add a test, so it won't be reintroduced in the future

**Pull request merged**